index.js

```diff
@@ -210,10 +210,10 @@ module.exports = class Hypercore extends EventEmitter {
     // but we only do this to validate the keypair to help catch bugs so yolo
     if (this.key && keyPair) keyPair.publicKey = this.key
 
```

Comment:
if `opts.sign` is set can you auto wrap into the new auth interface? for backwards compat (we can remove that later)